plugin/plugin.go

```diff
@@ -0,0 +1,249 @@
+// Copyright 2020 the Drone Authors. All rights reserved.
+// Use of this source code is governed by the Blue Oak Model License
+// that can be found in the LICENSE file.
+
+package plugin
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+type Args struct {
+	FlywayEnvPluginArgs
+	Level string `envconfig:"PLUGIN_LOG_LEVEL"`
+}
+
+type FlywayEnvPluginArgs struct {
+	DriverPath      string `envconfig:"PLUGIN_DRIVER_PATH"`
+	FlywayCommand   string `envconfig:"PLUGIN_FLYWAY_COMMAND"`
+	Locations       string `envconfig:"PLUGIN_LOCATIONS"`
+	CommandLineArgs string `envconfig:"PLUGIN_COMMAND_LINE_ARGS"`
+	Url             string `envconfig:"PLUGIN_URL"`
+	UserName        string `envconfig:"PLUGIN_USERNAME"`
+	Password        string `envconfig:"PLUGIN_PASSWORD"`
+	IsDryRun        string `envconfig:"PLUGIN_IS_DRY_RUN"`
+}
+
+type FlywayPlugin struct {
+	InputArgs         *Args
+	IsMultiFileUpload bool
+	ProcessingInfo
+}
+
+func (a *Args) ToStr() string {
+	jsonData, err := json.MarshalIndent(a, "", "  ")
+	if err != nil {
+		log.Printf("Error marshalling Args to JSON: %v", err)
+		return ""
+	}
+	return string(jsonData)
+}
+
+type ProcessingInfo struct {
+	ExecCommand         string
+	Env                 string
+	CommandSpecificArgs string
+}
+
+func Exec(ctx context.Context, args Args) (FlywayPlugin, error) {
+	plugin := FlywayPlugin{}
+
+	err := plugin.Init(&args)
+	if err != nil {
+		return plugin, err
+	}
+	defer func(p FlywayPlugin) {
+		err := p.DeInit()
+		if err != nil {
+			LogPrintln("Error in DeInit: " + err.Error())
+		}
+	}(plugin)
+
+	err = plugin.ValidateAndProcessArgs(args)
+	if err != nil {
+		return plugin, err
+	}
+
+	err = plugin.DoPostArgsValidationSetup(args)
+	if err != nil {
+		return plugin, err
+	}
+
+	err = plugin.Run()
+	if err != nil {
+		return plugin, err
+	}
+
+	return plugin, nil
+}
+
+func (p *FlywayPlugin) Init(args *Args) error {
+	p.InputArgs = args
+	return nil
+}
+
+func (p *FlywayPlugin) DeInit() error {
+	return nil
+}
+
+func (p *FlywayPlugin) ValidateAndProcessArgs(args Args) error {
+	err := p.IsCommandValid()
+	if err != nil {
+		LogPrintln(p, err.Error())
+		return err
+	}
+	return nil
+}
+
+func (p *FlywayPlugin) DoPostArgsValidationSetup(args Args) error {
+	if args.FlywayCommand == CleanCommand {
+		if !strings.Contains(args.CommandLineArgs, "-cleanDisabled") {
+			p.CommandSpecificArgs = "-cleanDisabled=false" + " "
+		}
+	}
+	return nil
+}
+
+func (p *FlywayPlugin) Run() error {
+	var stdoutBuf, stderrBuf bytes.Buffer
+	var err error
+
+	p.ExecCommand = p.GetExecArgsStr()
+	fmt.Println("Command: ", p.ExecCommand)
```

Comment:
```suggestion
	logrus.Infof("Executing command: %s", p.ExecCommand)
```

use logrus for consistency